website/content/v1.2/advanced/talos-api-access-from-k8s.md

```diff
@@ -0,0 +1,157 @@
+---
+title: "Talos API access from Kubernetes"
+description: "How to access Talos API from within Kubernetes."
+aliases:
+  - ../guides/talos-api-access-from-k8s
+---
+
+In this guide, we will enable the Talos feature to access the Talos API from within Kubernetes.
+
+## Enabling the Feature
+
+Edit the machine configuration to enable the feature, specifying the Kubernetes namespaces from which Talos API
+can be accessed and the allowed Talos API roles.
+
+```bash
+talosctl -n 172.20.0.2 edit machineconfig
+```
+
+Configure the `kubernetesTalosAPIAccess` like the following:
+
+```yaml
+spec:
+  machine:
+    features:
+      kubernetesTalosAPIAccess:
+        enabled: true
+        allowedRoles:
+          - os:reader
+        allowedKubernetesNamespaces:
+          - default
+```
+
+## Injecting Talos ServiceAccount into manifests
+
+Create the following manifest file `deployment.yaml`:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: talos-api-access
+spec:
+  selector:
+    matchLabels:
+      app: talos-api-access
+  template:
+    metadata:
+      labels:
+        app: talos-api-access
+    spec:
+      containers:
+        - name: talos-api-access
+          image: alpine:3
+          command:
+            - sh
+            - -c
+            - |
+              wget -O /usr/local/bin/talosctl https://github.com/siderolabs/talos/releases/download/<talos version>/talosctl-linux-amd64
+              chmod +x /usr/local/bin/talosctl
+              while true; talosctl -n 172.20.0.2 version; do sleep 1; done
+```
+
+**Note:** make sure that you replace the IP `172.20.0.2` with a valid Talos node IP.
+
+Use `talosctl kube-inject` command to inject the Talos ServiceAccount into the manifest.
+
+```bash
+talosctl kube-inject -f deployment.yaml > deployment-injected.yaml
+```
+
+Inspect the generated manifest:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  creationTimestamp: null
+  name: talos-api-access
+spec:
+  selector:
+    matchLabels:
+      app: talos-api-access
+  strategy: {}
+  template:
+    metadata:
+      creationTimestamp: null
+      labels:
+        app: talos-api-access
+    spec:
+      containers:
+      - command:
+        - sh
+        - -c
+        - |
+          wget -O /usr/local/bin/talosctl https://github.com/siderolabs/talos/releases/download/<talos version>/talosctl-linux-amd64
+          chmod +x /usr/local/bin/talosctl
+          while true; talosctl -n 172.20.0.2 version; do sleep 1; done
+        image: alpine:3
+        name: talos-api-access
+        resources: {}
+        volumeMounts:
+        - mountPath: /var/run/secrets/talos.dev
+          name: talos-secrets
+      tolerations:
+      - operator: Exists
+      volumes:
+      - name: talos-secrets
+        secret:
+          secretName: talos-api-access-talos-secrets
+status: {}
+---
+apiVersion: talos.dev/v1alpha1
```

Comment:
since we are doing a serviceaccount, couldn't we skip doing volume mounts and use `serviceAccountName` in the pod spec?

Reply:
We do not create k8s serviceaccounts, we have our own different serviceaccount with `serviceaccounts.talos.dev` type (easier if we call them "Talos Service Accounts"). We just provision secrets based on `serviceaccounts.talos.dev` custom resources.

Reply:
To further clarify, I did not name the CRD `talosserviceaccounts` because `talos` is redundant - it is also in the group name `talos.dev`.